src/pvsite_forecast.py

```diff
@@ -17,34 +17,76 @@
 import pytz
 
 # Penalty Calculator
-def calculate_penalty(df, capacity_kw=250000):
-    # Deviation between actual and forecast (in kW)
+# Give uuids for region + asset type + parse in the capcity from the site
+# Penalty Calculator
+def calculate_penalty(df, region, asset_type, capacity_kw):
+    """
+    Calculate penalties dynamically based on region, asset type, and capacity.
+    """
+    # Define penalty bands for combinations of region and asset type
+def calculate_penalty(df, region, asset_type, capacity_kw):
+    """
+    Calculate penalties dynamically based on region, asset type, and capacity.
+    """
+    # Define penalty bands for combinations of region and asset type
+    penalty_bands = {
+        ("Rajasthan", "solar"): [
+            (10, 15, 0.1),
+            (15, None, 1.0),
+        ],
+        ("Madhya Pradesh", "wind"): [
+            (10, 20, 0.25),
+            (20, 30, 0.5),
+            (30, None, 0.75),
+        ],
+        ("Gujarat", "solar"): [
+            (7, 15, 0.25),
+            (15, 23, 0.5),
+            (23, None, 0.75),
+        ],
+        ("Gujarat", "wind"): [
+            (12, 20, 0.25),
+            (20, 28, 0.5),
+            (28, None, 0.75),
+        ],
+        ("Karnataka", "solar"): [
+            (10, 20, 0.25),
+            (20, 30, 0.5),
+            (30, None, 0.75),
+        ],
+    }
```

Comment:
nice, i like this setup, i fyou could add a comment to say what the 1s, 2nd, and 3rd number mean, that would be great

Reply:
thanks, could you add what the units are too,